bob/src/hw_metrics_collector.rs

```diff
@@ -242,52 +294,105 @@ impl DiskStatCollector {
         }
 
         DiskStatCollector {
-            iostat_avl: true,
+            procfs_avl: true,
             disk_metric_data,
+            uptime: DiffContainer::new(),
+            reads: DiffContainer::new(),
+            writes: DiffContainer::new(),
+            read_time: DiffContainer::new(),
+            write_time: DiffContainer::new(),
         }
     }
 
     fn collect_and_send_metrics(&mut self) -> Result<(), CommandError> {
-        let iostat_lines = self.collect_iostat()?;
-        for i in iostat_lines {
+        let uptime = self.uptime()?;
+        let elapsed = self.uptime.diff(uptime);
```

Comment:
We don't need to parse `uptime` file to get elapsed time. It is simpler and faster to use `Instant` and `Duration` types